llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -1301,26 +1301,17 @@ bool NVPTXDAGToDAGISel::tryLDG(MemSDNode *LD) {
   return true;
 }
 
+unsigned NVPTXDAGToDAGISel::getFromTypeWidthForLoad(const MemSDNode *Mem) {
+  EVT MemVT = Mem->getMemoryVT();
+  auto ElementBitWidth = MemVT.getSizeInBits() / (Mem->getNumValues() - 1);
```

Comment:
Nice, we should use this everywhere we're using `getLoadStoreVectorNumElts` and maybe convert that function to `getStoreVectorNumElts`